src/construction.jl

```diff
@@ -1,18 +1,24 @@
 "Load all Arrow input data to SubDataFrames that are filtered for used IDs"
 function load_data(config::Dict, starttime::DateTime, endtime::DateTime)
-    node = DataFrame(read_table(config["node"]))
-    edge = DataFrame(read_table(config["edge"]))
-    state = DataFrame(read_table(config["state"]))
-    static = DataFrame(read_table(config["static"]))
-    profile = DataFrame(read_table(config["profile"]))
-    forcing = DataFrame(read_table(config["forcing"]))
+    # Load data and validate schema + rows for required field tyopes and values
+    node = read_table(config["node"]; schema = NodeV1SchemaVersion)
+    edge = read_table(config["edge"]; schema = EdgeV1SchemaVersion)
+    state = read_table(config["state"]; schema = StateV1SchemaVersion)
+    static = read_table(config["static"]; schema = StaticV1SchemaVersion)
+    profile = read_table(config["profile"]; schema = ProfileV1SchemaVersion)
+    forcing = read_table(config["forcing"]; schema = ForcingV1SchemaVersion)
+
+    # Validate consistency in the data
+    @assert is_consistent(node, edge, state, static, profile, forcing) "Data is not consistent"
```

Comment:
Legolas throws `ArgumentError`s. Perhaps we should not use assertions, see the docstring for `@assert`, "Warning: An assert might be disabled at various optimization levels."

I know I also used them already, but what do you think?

Reply:
I think it's fine. The warning has been there for a long time now, but asserts are always active. There is a PR introducing a `@check` that will never be disabled (https://github.com/JuliaLang/julia/pull/41342), whereas `@assert` could.

Reply:
Ok let's keep the asserts, we can change them if it becomes an issue.